packages/sdk/Dockerfile

```
@@ -286,14 +287,23 @@ apt-get install -y --no-install-recommends \
     libslirp0 \
     locales \
     lua5.4 \
-    nodejs \
-    npm \
     squashfs-tools \
     xxd \
     xz-utils
 rm -rf /var/lib/apt/lists/*
 EOF
 
+# Install nvm and node
+ENV NVM_DIR=/root/.nvm
+RUN <<EOF
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.3/install.sh | bash
```

I don't like this piping a script from the internet to shell.

We could improve this by using the git commit hash in the url, at least we have reproducibility without having to trust nvm release process but still trusting GitHub.
```suggestion
curl --proto '=https' --tlsv1.2 -o- https://raw.githubusercontent.com/nvm-sh/nvm/977563e97ddc66facf3a8e31c6cff01d236f09bd/install.sh | bash
```

The hash could be on a build arg if you will.


The ultimate process here would to git clone and verify the tag with gpg, but I won't complain if you don't.

---

I don’t like piping to shell either, but I don’t like to deviate too much from the documented standard way of installing.

what is a good middle ground in this case?

---

We need to run the `install.sh`, how we obtain and verify its integrity is up to us.

piping to shell is a convenient way, not the recommended way, I think.

---

My suggestion was already a middle ground, by using the hash in the URL, we make sure there wasn't content modification, since tags can e changed.

And we trust GitHub is not messing around.

---

Every nvm installation instruction pipe to shell, that's what I meant by "recommended".
Updated with your suggestion.